sstables/mx/reader.cc

```diff
@@ -350,350 +350,461 @@ class data_consume_rows_context_m : public data_consumer::continuous_data_consum
     }
 private:
     proceed_generator do_process_state(bool skip_partition) {
-        // switch (_state) {
-        // case state::PARTITION_START:
-        partition_start_label:
-        if (!skip_partition) {
-            _is_first_unfiltered = true;
-            if (read_short_length_bytes(*_processing_data, _pk) != read_status::ready) {
-                _state = state::DELETION_TIME;
-                co_yield consumer_m::proceed::yes;
-            }
-        // case state::DELETION_TIME:
-            if (read_32(*_processing_data) != read_status::ready) {
-                _state = state::DELETION_TIME_2;
-                co_yield consumer_m::proceed::yes;
-            }
-        // case state::DELETION_TIME_2:
-            if (read_64(*_processing_data) != read_status::ready) {
-                _state = state::DELETION_TIME_3;
-                co_yield consumer_m::proceed::yes;
-            }
-        // case state::DELETION_TIME_3: {
-            deletion_time del;
-            del.local_deletion_time = _u32;
-            del.marked_for_delete_at = _u64;
-            auto ret = _consumer.consume_partition_start(key_view(to_bytes_view(_pk)), del);
-            // after calling the consume function, we can release the
-            // buffers we held for it.
-            _pk.release();
-            _state = state::FLAGS;
-            if (ret == consumer_m::proceed::no) {
-                co_yield consumer_m::proceed::no;
-            }
-        }
-        skip_partition = false;
-        // case state::FLAGS:
-        while(true) {
-            _liveness = {};
-            _row_tombstone = {};
-            _row_shadowable_tombstone = {};
-            if (read_8(*_processing_data) != read_status::ready) {
-                _state = state::FLAGS_2;
-                co_yield consumer_m::proceed::yes;
-            }
-        // case state::FLAGS_2:
-            _flags = unfiltered_flags_m(_u8);
-            if (_flags.is_end_of_partition()) {
-                _state = state::PARTITION_START;
-                if (_consumer.consume_partition_end() == consumer_m::proceed::no) {
-                    co_yield consumer_m::proceed::no;
+        while (true) {
+            switch (_state) {
```

Comment:
So, here, finally, is the missing switch I wondered about.
So how did this code work before this patch? I am guessing it didn't.
Patch series should be bisectable - this is important for "git bisect" but also to human reviewers like me who don't understand all these small incremental changes, which look wrong because they are (apparently) wrong - the previous patches were missing this while() and switch(), and because of that were wrong.

Reply:
It's probable that the code didn't work for some patches, sorry for that. All patches in the new PR should work however.